src/extension/provider/annotations.ts

```diff
@@ -49,41 +45,7 @@ export class AnnotationsProvider implements NotebookCellStatusBarItemProvider {
   }
 
   public async toggleCellAnnotations(cell: NotebookCell): Promise<void> {
-    const annotationsExists = cell.outputs.find((o) =>
-      o.items.find((oi) => oi.mime === OutputType.annotations)
-    )
-
-    let exec
-    try {
-      exec = await this.kernel.createCellExecution(cell)
-      exec.start(Date.now())
-
-      if (annotationsExists) {
-        exec.clearOutput()
-        return
-      }
-
-      const json = <CellOutputPayload<OutputType.annotations>>{
-        type: OutputType.annotations,
-        output: {
-          annotations: getAnnotations(cell),
-          validationErrors: validateAnnotations(cell)
-        },
-      }
-      await replaceOutput(exec, [
-        new NotebookCellOutput([
-          NotebookCellOutputItem.json(json, OutputType.annotations),
-          NotebookCellOutputItem.json(json),
-        ]),
-      ])
-    } catch (e: any) {
-      if (e.message.toString().includes('controller is NOT associated')) {
-        return this.handleNotebookKernelSelection()
-      }
-      window.showErrorMessage(e.message)
-    } finally {
-      exec?.end(true)
-    }
+    await (await this.kernel.getCellOutputs(cell)).toggleOutput(OutputType.annotations)
```

Comment:
maybe easier to read:
```suggestion
    const outputCell = await this.kernel.getCellOutputs(cell)
    await outputCell.toggleOutput(OutputType.annotations)
```